examples/fishnet_api/main.py

```diff
@@ -286,140 +348,150 @@ async def request_execution(
+        if (rec.datasetID in ds_ids and rec.status == ExecutionStatus.PENDING):
+            rec.status = ExecutionStatus.DENIED
+            requests.append(rec.upsert())
+    # parellel processed
+    await asyncio.gather(*requests)
+    return permission_records
 
 
 @app.put("/datasets/{dataset_id}/available/{available}")
-async def set_dataset_available(dataset_id: str, available: bool):
+async def set_dataset_available(dataset_id: str, available: bool) -> List[Dataset]:
     """
-    Set a dataset to be available or not. This will also update the status of all executions that are waiting for
-    permission on this dataset.
+    Set a dataset to be available or not. This will also update the status of all
+    executions that are waiting for permission on this dataset.
     """
     # Check signature to match with owner's
     # This signature will be implemented by Mike
     requests = []
     resp = await Dataset.fetch(dataset_id)
     # Handled the case when the dataset is not found
-    if resp:
-        dataset = resp[0]
-        dataset.available = available
-        requests.append(dataset.upsert())
-
-        # Get all timeseries in the dataset and set them to available or not
-        ts_list = await Timeseries.fetch(dataset.timeseriesIDs)
-        if ts_list:
-            ts_record = ts_list[0]
-            # check if the timeseries is updated or not and only update the timeseries when it is needed
-            if ts_record.available != available:
-                ts_record.available = available
-                requests.append(ts_record.upsert())
-            else:
-                return {"error": "Record is already Updated"}
-
-            # Get all executions that are waiting for this dataset (status == PENDING) and update their status to DENIED
-            execution = await Execution.fetch(dataset_id)
-            if execution:
-                for execution_rec in execution:
-                    if execution_rec.status == ExecutionStatus.PENDING:
-                        execution_rec.status = ExecutionStatus.DENIED
-                        requests.append(execution_rec.upsert())
-
-                # executed all requests in parallel
-                await asyncio.gather(requests)
-                return {"Success": "Dataset availability has been set successfully "}
-            else:
-                return {"Execution": "No Execution found "}
-        else:
-            return {"Timeseries": "No Timeseries found"}
+    if not resp:
+        raise HTTPException(status_code=424, detail="No Dataset found")
+    dataset = resp[0]
+    dataset.available = available
+
+    requests.append(dataset.upsert())
+
+    # Get all timeseries in the dataset and set them to available or not
+    ts_list = await Timeseries.fetch(dataset.timeseriesIDs)
+    if not ts_list:
+        raise HTTPException(status_code=424, detail="No Timeseries found")
+    ts_record = ts_list[0]
```

Comment:
you are only checking for the first time series, but you need to check for all received time series